plugin-cache/index.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+var EventEmitter = require('events').EventEmitter;
+var Q = require('q');
+var util = require('util');
+
+var getList = require('./src/get');
+var updateList = require('./src/update');
+
+var Logger = process.env.Logger || {
+  log: console.log.bind(console),
+  error: console.error.bind(console)
+};
+
+var Q = process.env.Q || Q;
+
+var Cache = function () {
+  this.list = [];
+
+  this.q = Q;
+
+  this.get = getList.bind(this);
```

Comment:
Class property should be define in the same file if they depend on the class (like in this case). This make it easier to understand the context when looking at these function.

This is a mistake we've done in `yeoman-generator`, it'd be nicer to not repeat it.